pkg/jsonnetfile/jsonnetfile.go

```diff
@@ -19,44 +19,84 @@ import (
 	"io/ioutil"
 	"os"
 
+	v2 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v2"
+	v3 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v3"
+	depsv3 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v3/deps"
 	"github.com/pkg/errors"
-
-	"github.com/jsonnet-bundler/jsonnet-bundler/spec"
-	"github.com/jsonnet-bundler/jsonnet-bundler/spec/deps"
 )
 
 const (
 	File     = "jsonnetfile.json"
 	LockFile = "jsonnetfile.lock.json"
 )
 
-var ErrNoFile = errors.New("no jsonnetfile")
+var (
+	ErrNoFile   = errors.New("no jsonnetfile")
+	ErrUpdateJB = errors.New("jsonnetfile version unknown, update jb")
+)
 
 // Load reads a jsonnetfile.(lock).json from disk
-func Load(filepath string) (spec.JsonnetFile, error) {
+func Load(filepath string) (v3.JsonnetFile, error) {
 	bytes, err := ioutil.ReadFile(filepath)
 	if err != nil {
-		return spec.New(), err
+		return v3.New(), err
 	}
 
 	return Unmarshal(bytes)
 }
 
 // Unmarshal creates a spec.JsonnetFile from bytes. Empty bytes
 // will create an empty spec.
-func Unmarshal(bytes []byte) (spec.JsonnetFile, error) {
-	m := spec.New()
+func Unmarshal(bytes []byte) (v3.JsonnetFile, error) {
+	m := v3.New()
+
 	if len(bytes) == 0 {
 		return m, nil
 	}
-	if err := json.Unmarshal(bytes, &m); err != nil {
-		return m, errors.Wrap(err, "failed to unmarshal file")
+
+	versions := struct {
+		Version float64 `json:"version"`
+	}{}
+
+	err := json.Unmarshal(bytes, &versions)
+	if err != nil {
+		return m, err
 	}
-	if m.Dependencies == nil {
-		m.Dependencies = make(map[string]deps.Dependency)
+
+	if versions.Version > 3 {
```

Comment:
What about moving this magic constant `3` up and use a constant (`currentlySupportedVersion`) ?